Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -124,9 +124,31 @@ class BrowserViewController: UIViewController {
     }
 
     override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
         updateToolbarState()
         showToolbars(animated: false)
-        super.traitCollectionDidChange(previousTraitCollection)
+    }
+
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+
+        /**
+        *  WKWebView looks like it has a bug where it doesn't invalidate it's visible area when the user
+        *  performs a device rotation. Since scrolling calls
+        *  _updateVisibleContentRects (https://github.com/WebKit/webkit/blob/master/Source/WebKit2/UIProcess/API/Cocoa/WKWebView.mm#L1430)
+        *  this method nudges the web view's scroll view by a single pixel to force it to invalidate.
+        */
+        coordinator.animateAlongsideTransition({ context in
+            if let tab = self.tabManager.selectedTab {
+                let contentOffset = tab.webView.scrollView.contentOffset
+                tab.webView.scrollView.setContentOffset(CGPoint(x: contentOffset.x, y: contentOffset.y + 1), animated: true)
+            }
+        }, completion: { context in
+            if let tab = self.tabManager.selectedTab {
+                let contentOffset = tab.webView.scrollView.contentOffset
+                tab.webView.scrollView.setContentOffset(CGPoint(x: contentOffset.x, y: contentOffset.y - 1), animated: true)
```

Comment:
If you want, you could factor out the `if let tab = ...` and the `let contentOffset = ...` to outside the `coordinator.animateAlongsideTransition` so you don't have to duplicate them in each block. (If you did that, I guess you'd also change `contentOffset.y - 1` -> `contentOffset.y` since it would still hold the original value).